src/contracts/helpers/DemoLendingProtocol.sol

```diff
@@ -0,0 +1,114 @@
+    IChainlinkFeed public chainlinkFeed;
+
+    mapping(address account => Position) public positions;
+
+    error PositionNotLiquidatable();
+    error PositionNotFound();
+    error PositionAlreadyExists();
+
+    event Deposit(address indexed account, uint256 amount, uint256 liquidationPrice);
+    event Withdraw(address indexed account, uint256 amount);
+    event Liquidation(address indexed account, address indexed recipient, uint256 amount);
+
+    constructor(address depositToken) Ownable() {
+        DEPOSIT_TOKEN = IERC20(depositToken);
+    }
+
+    // ---------------------------------------------------- //
+    //                    Setup Functions                   //
+    // ---------------------------------------------------- //
+
+    // Deposits `amount` of deposit token into caller's position, setting the liquidation price to `liquidationPrice`.
+    function deposit(uint256 amount, uint256 liquidationPrice) external {
+        Position memory position = positions[msg.sender];
+
+        if (position.amount > 0) revert PositionAlreadyExists();
+
+        DEPOSIT_TOKEN.safeTransferFrom(msg.sender, address(this), amount);
+        positions[msg.sender] = Position(amount, liquidationPrice);
+
+        emit Deposit(msg.sender, amount, liquidationPrice);
+    }
+
+    // Withdraws funds in caller's position if not liquidated, and deletes the position.
+    function withdraw() external {
+        Position memory position = positions[msg.sender];
+
+        if (position.amount == 0) revert PositionNotFound();
+
+        DEPOSIT_TOKEN.safeTransfer(msg.sender, position.amount);
+        delete positions[msg.sender];
+
+        emit Withdraw(msg.sender, position.amount);
+    }
+
+    // ---------------------------------------------------- //
+    //                    Solver Functions                  //
+    // ---------------------------------------------------- //
+
+    // Called by a solver. If targeted account's position is liquidatable, the position amount is sent to the
+    // `recipient` address. If position is not liquidatable, reverts.
+    function liquidate(address account, address recipient) external {
+        Position memory position = positions[account];
+
+        if (position.amount == 0) revert PositionNotFound();
+        if (position.liquidationPrice < uint256(chainlinkFeed.latestAnswer())) revert PositionNotLiquidatable();
+
+        // Send liquidated amount to solver's recipient, and delete the account's position
+        DEPOSIT_TOKEN.safeTransfer(recipient, position.amount);
+        delete positions[account];
+
+        emit Liquidation(account, recipient, position.amount);
+    }
+
+    // Helper view function for solvers to check if a position is liquidatable
+    function isLiquidatable(address account) external view returns (bool) {
+        Position storage position = positions[account];
+        return (position.amount > 0) && (uint256(chainlinkFeed.latestAnswer()) <= position.liquidationPrice);
```

Comment:
```suggestion
        return (position.amount > 0) && (uint256(chainlinkFeed.latestAnswer()) < position.liquidationPrice);
```

Reply:
Cool, will standardize on `(liqPrice < oraclePrice) => liquidation allowed`. Will need another change in `liquidate()` function I think.